test/unittest/tests/test_session_config.cpp

```diff
@@ -0,0 +1,66 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <catch2/catch.hpp>
+
+#include "wt_internal.h"
+#include "wrappers/mock_session.h"
+
+/*
+ * Check a basic configuration that sets and clears a flag.
+ */
+#define FLAG_TEST(config_param, flag)                                                    \
+    TEST_CASE(config_param, "[session_config]")                                          \
+    {                                                                                    \
+        std::shared_ptr<MockSession> session_mock = MockSession::buildTestMockSession(); \
+        WT_SESSION_IMPL *session = session_mock->getWtSessionImpl();                     \
+        session->flags = 0;                                                              \
+        REQUIRE(__ut_session_config_int(session, config_param "=true") == 0);            \
+        REQUIRE(F_ISSET(session, flag));                                                 \
+        REQUIRE(__ut_session_config_int(session, config_param "=false") == 0);           \
+        REQUIRE(!F_ISSET(session, flag));                                                \
+        REQUIRE(__ut_session_config_int(session, config_param "=true") == 0);            \
+        REQUIRE(__ut_session_config_int(session, "") == 0);                              \
```

Comment:
I believe new line 28 is intended to test that __session_config_int() returns 0 if config does not match any of the config parameters it parses.I suggest to add another test that mentions a parameter not parsed by __session_config_int(). Perhaps add
REQUIRE(__ut_session_config_int(session, "not_a_real_parameter=false") == 0);